<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GdbConfigPage</class>
 <widget class="QWidget" name="GdbConfigPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>860</width>
    <height>466</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>sdsGdbConfigPage</string>
  </property>
  <widget class="QGroupBox" name="gdbGroupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>839</width>
     <height>148</height>
    </rect>
   </property>
   <property name="title">
    <string>Gdb设置</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="2" column="0">
     <widget class="QCheckBox" name="gdbAsyncModeCheckBox">
      <property name="toolTip">
       <string>Turn gdb 'async' mode on or off.</string>
      </property>
      <property name="text">
       <string>GDB asycn 模式</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLineEdit" name="gdbProgramLineEdit">
      <property name="toolTip">
       <string>gdb调试器可执行文件的位置</string>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <widget class="QToolButton" name="gdbProgramToolButton">
      <property name="toolTip">
       <string>选择gdb位置</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset resource="../SeeDataStructureRc.qrc">
        <normaloff>:/resources/RelaxLightIcons/document-open.svg</normaloff>:/resources/RelaxLightIcons/document-open.svg</iconset>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="gdbArgumentsLabel">
      <property name="text">
       <string>GDB 运行参数</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLineEdit" name="gdbArgumentsLineEdit">
      <property name="toolTip">
       <string>传递给gdb调试器的参数</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="gdbProgramLabel">
      <property name="text">
       <string>GDB 程序位置</string>
      </property>
     </widget>
    </item>
    <item row="2" column="2">
     <widget class="QCheckBox" name="gdbHandleTerminateExceptionCheckBox">
      <property name="toolTip">
       <string>Handle terminating exceptions when calling functions.</string>
      </property>
      <property name="text">
       <string>处理 terminating-exception</string>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QToolButton" name="gdbTestToolButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>10</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="sizeIncrement">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>测试gdb是否正常</string>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="text">
       <string>测试GDB</string>
      </property>
      <property name="icon">
       <iconset resource="../SeeDataStructureRc.qrc">
        <normaloff>:/resources/icons/test.svg</normaloff>:/resources/icons/test.svg</iconset>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>170</y>
     <width>839</width>
     <height>281</height>
    </rect>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt; font-weight:600;&quot;&gt;在这里指定你的gdb可执行文件目录，并指定参数。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Noto Sans'; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Noto Sans'; font-size:10pt;&quot;&gt;本项目依赖于gdb中内置的&amp;quot;mi&amp;quot;解释器. &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Noto Sans'; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Noto Sans'; font-size:10pt;&quot;&gt;Async，也就是异步模式允许所有的gdb动作在后台运行。这样你就可以打断程序的运行。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Noto Sans'; font-size:10pt;&quot;&gt;当请求gdb调用程序中的函数时，可能会发生终止异常terminating-exception。如果函数抛出异常，gdb可以通过抑制异常或允许程序终止来处理它。可以参考:&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'Noto Sans'; font-size:10pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace'; font-size:10pt; color:#24292f; background-color:transparent;&quot;&gt;    unwind-on-terminating-exception on|off&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace'; font-size:10pt; color:#24292f;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace'; font-size:10pt; color:#24292f; background-color:transparent;&quot;&gt;点击测试按钮&lt;/span&gt;&lt;img src=&quot;:/resources/icons/test.svg&quot; width=&quot;20&quot; height=&quot;20&quot; /&gt;&lt;span style=&quot; font-family:'ui-monospace,SFMono-Regular,SF Mono,Menlo,Consolas,Liberation Mono,monospace'; font-size:10pt; color:#24292f; background-color:transparent;&quot;&gt;测试您指定的参数是否能正确启动gdb&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
 </widget>
 <resources>
  <include location="../SeeDataStructureRc.qrc"/>
 </resources>
 <connections/>
</ui>
